src/Microsoft.AspNetCore.Sockets.Http/SocketRouteBuilder.cs

```diff
@@ -28,6 +28,7 @@ public void MapSocket(string path, HttpSocketOptions options, Action<ISocketBuil
             socketConfig(socketBuilder);
             var socket = socketBuilder.Build();
             _routes.MapRoute(path, c => _dispatcher.ExecuteAsync(c, options, socket));
+            _routes.MapRoute(path + "/negotiate", c => _dispatcher.ExecuteNegotiateAsync(c, options));
```

Comment:
Hmmm

Reply:
What are your thoughts

Reply:
The `HttpConnectionDispatcher` doesn't know the path the user asked to map the socket too, so this is the cleanest way to do this. I actually like it better anyway, the dispatcher shouldn't have to be doing routing things.